File: atividade_8/java/Peca.java
package java;

public class Peca {
    private String tipo;
    private String cor;
    private String posicao;

    public Peca(String tipo, String cor, String posicao) {
        this.tipo = tipo;
        this.cor = cor;
        this.posicao = posicao;
    }

    public void moverPeca(String pos) {
        this.posicao = pos;
    }

    public boolean capturaPeca(String type, String posicaoOrigem, String posDestino) {
        if (!vericaPosicao(type, posicaoOrigem, posDestino)) {
            return false;
        }
        moverPeca(posDestino);
        return true;
    }

    public boolean vericaPosicao(String type, String posicaoOrigem, String posDestino) {
        if (type.equals("cavalo") && !(posicaoOrigem.equals(posDestino))) {
            return true;
        }
        return false;
    }

    @Override
    public String toString() {
        StringBuilder builder = new StringBuilder();
        builder.append("Peca [tipo=").append(tipo).append(", cor=").append(cor).append(", posicao=").append(posicao)
                .append("]");
        return builder.toString();
    }
}

File: atividade_8/C/peca.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct Peca {
    char tipo[20];
    char cor[20];
    char posicao[3];
};

void inicializarPeca(struct Peca *p, char tipo[], char cor[], char posicao[]) {
    strcpy(p->tipo, tipo);
    strcpy(p->cor, cor);
    strcpy(p->posicao, posicao);
}

void moverPeca(struct Peca *p, char pos[]) {
    strcpy(p->posicao, pos);
}

int capturaPeca(struct Peca *p, char type[], char posOrigem[], char posDestino[]) {
    if (!verificaPosicao(type, posOrigem, posDestino)) {
        return 0;
    }
    moverPeca(p, posDestino);
    return 1;
}

int verificaPosicao(char type[], char posOrigem[], char posDestino[]) {
    if (strcmp(type, "cavalo") == 0 && strcmp(posOrigem, posDestino) != 0) {
        return 1;
    }
    return 0;
}